### PR TITLE
Use set to do the compare

### PR DESCRIPTION
**Pull request by kevinzhow**

### CODE REVIEW

**Comment by AFutureD** [comment]
Please, rebase to the main branch (13a8696), package name in tests has fixed.

**Comment by kevinzhow** [comment]
> Please, rebase to the main branch ([13a8696](https://github.com/Myoland/VaporAuth/commit/13a86961690e09463c70f18853075a504cb8255a)), package name in tests has fixed.

Done

File: Sources/VaporAuth/Claims/ScopeClaim.swift [hunk 1]
```diff
@@ -42,25 +43,13 @@ extension ScopeClaim: ExpressibleByArrayLiteral {
...
-        let requiredWrappers = lhs.wrapper
-        let carriedWrappers = rhs.wrapper
-        
-        for require in requiredWrappers {
-            guard carriedWrappers.contains(where: { require <= $0 }) else  {
-                return false
-            }
-        }
-        return true
+        return Set(lhs.value).isStrictSubset(of: Set(rhs.value))
```
Comment (AFutureD) [inline Sources/VaporAuth/Claims/ScopeClaim.swift:47]: [Set.isStrictSubset(of:)](https://developer.apple.com/documentation/swift/set/isstrictsubset(of:)-96vc3) only perform "=="  on each element.
In this method, we required every scope not only equal to one of the carried scopes, but also less than one.
So this part may not replace with set operation.

File: Sources/VaporAuth/Claims/ScopeClaim.swift [hunk 2]
```diff
@@ -42,25 +43,13 @@ extension ScopeClaim: ExpressibleByArrayLiteral {
...
     }
 }
 
 extension ScopeClaim: Guardable {
     public func hasAuth(required: ScopeClaim) -> Bool {
-        required == self || required < self
+        required <= self
```
Comment (AFutureD) [inline Sources/VaporAuth/Claims/ScopeClaim.swift:53]: This is acceptable.

**Comment by AFutureD** [comment]
I'm try to make `Scope` be optional.
And I'm not sure if the `Scope` only need `==`.

**Pull request closed without merging**